internal/access/provider.go

```diff
@@ -15,6 +15,17 @@ import (
 	"github.com/infrahq/infra/uid"
 )
 
+func isInternalProvider(c *gin.Context, requestedID uid.ID) (bool, error) {
+	db := getDB(c)
+
+	internalProvider, err := data.GetProvider(db, data.ByName(models.InternalInfraProviderName))
```

Comment:
You can use the `InfraProvider()` function here to get the internal provider simply